src/coreclr/jit/hwintrinsic.cpp

```diff
@@ -497,6 +497,14 @@ GenTree* Compiler::getArgForHWIntrinsic(var_types            argType,
             impPushOnStack(gtNewLclvNode(tmp, lvaGetRealType(tmp)), verMakeTypeInfo(argClass).NormaliseForStack());
         }
     }
+    else if (argType == TYP_REF)
```

Comment:
This is basically duplicating everything from the `else` path but with a slightly different assert.

Can we just merge the two paths?

Reply:
Are you suggesting to merge the asserts or making them more relaxed?

Reply:
Likely merging them.